pkg/frontend/querymiddleware/codec.go

```diff
@@ -226,6 +230,19 @@ func decodeOptions(r *http.Request, opts *Options) {
 	}
 }
 
+func decodeInstantQueryOptions(r *http.Request, opts *Options) {
```

Comment:
I wouldn't add a new function, but add the logic to the existing `decodeOptions()` which is expected to decode all options.

Reply:
But `decodeOptions()` is also used for range query requests: https://github.com/grafana/mimir/blob/main/pkg/frontend/querymiddleware/codec.go#L191

Do we want to also add support for this header on range requests? It wouldn't be used for anything, but it might be confusing. 
Maybe a better solution would be to add it directly in the `decodeInstantQueryRequest()`. Wdyt? 🙂

Reply:
We only have 1 Options struct (used both by range and instant queries), so I would just merge everything in `decodeOptions()`. Then based on each type of query, we'll practically look only at the supported options.